pkg/extensions/lint/lint.go

```diff
@@ -26,7 +34,57 @@ func NewLinter(config *config.LintConfig, log log.Logger) *Linter {
 	}
 }
 
-func (linter *Linter) CheckMandatoryAnnotations(repo string, manifestDigest godigest.Digest,
+func (linter *Linter) CheckMandatoryConditions(repo string, manifestDescriptor ispec.Descriptor,
+	imgStore storage.ImageStore,
+) (bool, error) {
+	passing := false
+
+	passing, err := linter.CheckMandatoryAnnotations(repo, manifestDescriptor, imgStore)
+	if err != nil || !passing {
+		return passing, err
+	}
+
```

Comment:
Since CheckArtifactIsLogo is to be used only for artifacts, another condition could be used here before calling the function, such as
if manifestDescriptor.MediaType == ispec.MediaTypeArtifactManifest {
		return linter.CheckArtifactIfLogo(repo, manifestDescriptor.Digest, imgStore)
	}